docker/docker-bake-rust-all.sh

```diff
@@ -6,11 +6,10 @@
 # You need to execute this from the repository root as working directory
 # E.g. docker/docker-bake-rust-all.sh
 
-set -e
+set -ex
 
 ## TODO(christian): add `--progress=plain` as soon as circleci supports a docker version that includes https://github.com/moby/buildkit/pull/2763
 export GIT_REV=$(git rev-parse --short=8 HEAD)
 export GIT_SHA1=$(git rev-parse HEAD)
 export BUILD_DATE="$(date -u +'%Y-%m-%dT%H:%M:%SZ')"
-export GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
```

Comment:
should we leave `GIT_BRANCH` here instead of specifying in the GH action to prevent relying on the platform too much? e.g. we can run this script from dev machines as well

Reply:
the reason I removed this is that when you run above command in github actions it actually returns `HEAD` instead of the branch name. This is probably because by default github actions does only a shallow checkout.
I'll look into making the docker build fully locally executable in another PR (there are a few other issues right now, such as lack of M1 support etc.).